### DISCUSSION TITLE
How does `workbench` work?

### DISCUSSION THREAD

**Discussion by nhedger** [orchestral/testbench#392] [Q&A]
Hey,

I'm trying to figure out how the `workbench` directory works and whether it's related to the skeleton mentioned in the docs. Because of the structure of this folder, I'm assuming it's some barebones Laravel application that I can run to interact with the package I'm building.

I have the ServiceProvider of my package defined in `src/ServiceProvider.php,` but it does seem to get registered in the workbench application. I tried registering the provider by adding it to the `providers` key in `testbench.yaml`, but it does not seem to work.

By the way, I have not yet set up any tests; I just wanted to see if I could load my service provider into the workbench.

Not sure if I'm going about this correctly, and I could use some help.

Thanks

**Comment by feryardiant** [comment]
Hi,

I'm also wondering kinda similar question regarding `workbench` directory. But my understanding so far is that `workbench` directory and its `testbench.yaml` would works only under `testbench serve` command or perhaps any other `testbench` commands except the `testbench package:test`

For instance, I have `testbench.yaml` like below

```yml
providers:
  - MyPackage\MyPackageServiceProvider
```

I can only see it's being loaded when I run `./vendor/bin/testbench serve`, but when I want to see it loadied when I run `./vendor/bin/testbench package:test` I'm still have to register it to `getPackageProviders()` method in my `TestCase`

```php
/**
 * @param  \Illuminate\Foundation\Application  $app
 */
protected function getPackageProviders($app): array
{
    return [
        MyPackage\MyPackageServiceProvider::class,
    ];
}
```

Thanks

**Reply by crynobone** [reply]
See https://packages.tools/testbench.html#autoloading-using-testbench-yaml
